cosmic/utils.py

```diff
@@ -61,19 +62,45 @@ def filter_bpp_bcm(bcm, bpp, method):
 
     for meth, use in method.items():
         if meth == 'mass_transfer_white_dwarf_to_co' and not use:
-            idx_mass_transfer_white_dwarf_to_co = bpp.loc[(bpp.kstar_1.isin([10,11,12,13,14])) &
-                                                          (bpp.kstar_2.isin([10,11,12])) &
-                                                          (bpp.evol_type == 3.0)].bin_num
-            bcm = bcm.loc[~bcm.bin_num.isin(idx_mass_transfer_white_dwarf_to_co)]
+            idx_save = bpp.loc[~(bpp.kstar_1.isin([10,11,12,13,14])) &
+                                (bpp.kstar_2.isin([10,11,12])) &
+                                (bpp.evol_type == 3.0)].bin_num.tolist()
+            bcm = bcm.loc[~bcm.bin_num.isin(idx_save)]
         elif (meth == 'select_final_state') and use:
             bcm = bcm.iloc[bcm.reset_index().groupby('bin_num').tphys.idxmax()]
         elif (meth == 'binary_state'):
             bcm = bcm.loc[bcm.bin_state.isin(use)]
+            
+            # CREATE A LIST OF BIN_NUM INDICES TO APPEND TO
+            bin_num_save = []
```

Comment:
`bin_num_save` looks great but it makes me think should we not have a grand `bin_num_save` list at the beginning of the filtering before we loop over all the methods and simply append to this list the `idx` for all binaries we are saving? so you have `idx_save` for `mass transfer....` and for `LISA_sources` etc. would it be smoother and easier to add new filters if we had a master `bin_num_save` list to append to instead of just in `binary_state`?

Reply:
I generally agree with this but am having a difficult time finding the right logic specifically for the mass transfer filtering and the select final state filtering.

The mass transfer filter is difficult because the logic is to throw out values based on some criteria, instead of retaining values based on that criteria. If we just do in the inverse, keep everything that _doesn't_ satisfy the criteria, then we will have to filter it out later on. 

With the select final state filter, we have to appeal to iloc to get the actual highest index in the bcm array to select the final time.

These complications lead me to think that it isn't just as simple as appending to a `bin_num_save` list for all the different types of filtering.